tests/pytest/core/test_session.py

```diff
@@ -0,0 +1,400 @@
+import time
+
+from django.urls import reverse
+import pytest
+
+from benefits.core import models, session
+
+
+@pytest.mark.django_db
+def test_active_agency_False(session_request):
+    session.update(session_request, agency=None)
+
+    assert not session.active_agency(session_request)
+
+
+@pytest.mark.django_db
+def test_active_agency_True(session_request):
+    agency = models.TransitAgency.objects.filter(active=True).first()
+    session.update(session_request, agency=agency)
+
+    assert session.active_agency(session_request)
+
+
+@pytest.mark.django_db
+def test_debug_default(session_request):
+    assert session._DEBUG not in session_request.session
+
+    debug = session.debug(session_request)
+
+    assert isinstance(debug, bool)
+    assert not debug
+
+
+@pytest.mark.django_db
+def test_did(session_request):
+    assert session._DID not in session_request.session
+
+    did = session.did(session_request)
+
+    assert did
+    assert isinstance(did, str)
+    assert session._DID in session_request.session
+
+
+@pytest.mark.django_db
+def test_eligibility_default(session_request):
+    assert session._ELIGIBILITY not in session_request.session
+    assert session.eligibility(session_request) is None
+
+
+@pytest.mark.django_db
+def test_eligibile_False(session_request):
+    assert session.eligibility(session_request) is None
+    assert not session.eligible(session_request)
+
+
+@pytest.mark.django_db
+def test_eligibile_True(session_request):
+    agency = models.TransitAgency.objects.first()
+    eligibility = agency.eligibility_types.first()
+
+    session.update(session_request, agency=agency, eligibility_types=[eligibility.name])
+
+    assert session.eligible(session_request)
+
+
+@pytest.mark.django_db
+def test_enrollment_token_default(session_request):
+    assert session._ENROLLMENT_TOKEN not in session_request.session
+    assert session.enrollment_token(session_request) is None
+    assert session._ENROLLMENT_TOKEN_EXP not in session_request.session
+    assert session.enrollment_token_expiry(session_request) is None
+
+
+@pytest.mark.django_db
+def test_enrollment_token_valid(session_request):
+    # valid token expiring in the far future
```

Comment:
Should we also have a test for an expired one?

Reply:
Done in [bcf5562](https://github.com/cal-itp/benefits/pull/474/commits/bcf556238953f05a18126742c0dac9789e41991b)